src/client/components/EditTeamModal.tsx

```diff
@@ -0,0 +1,134 @@
+                            <div className="col-12">
+                                <h2>Members</h2>
+                                <ul>
+                                    {props.members.map((member, i) => {
+                                        const newMember = newMembers.includes(member.account.email);
+                                        return (
+                                            <li
+                                                className={classnames("list-group-item d-flex justify-content-between align-items-center", {
+                                                    "list-group-item-success": newMember
+                                                })}
+                                                key={i}
+                                            >
+                                                {!newMember && <>{member.lastName}, {member.firstName} ({member.account.email})</>}
+                                                {newMember && <>{member.account.email}</>}
+                                                <Button
+                                                    // @ts-ignore
+                                                    size="sm"
+                                                    onClick={() => props.array.remove('members', i)}
+                                                >
+                                                    <FA name="minus" />
+                                                </Button>
+                                            </li>
+                                        )
+                                        })}
+
+                                    {props.members.length < MAX_TEAM_SIZE && (
+                                        <div className="list-group-item d-flex justify-content-between align-items-center">
+                                            <label htmlFor="newEmail" className="sd-form__label">Add a Teammate</label>
+                                            <input
+                                                name="newEmail"
+                                                className="sd-form__input-text"
+                                                type="email"
+                                                placeholder="triton@ucsd.edu"
+                                                value={newEmail}
+                                                onChange={event => setNewEmail(event.target.value)}
+                                            />
+                                            <Button
+                                                // @ts-ignore
+                                                size="sm"
+                                                style={{ marginLeft: 20 }}
+                                                onClick={() => {
+                                                    props.array.push('members', { account: { email: newEmail } });
```

Comment:
@subhankar-panda @nickthegroot  Type unsafe. Would be better to have arrays of `addMembers` and `removeMembers` storing relevant emails, rather than attempting to add/remove `TESCUser` elements to the members array.

Reply:
isnt mutation props just plain wrong too?